src/nbd_input.ml

```diff
@@ -0,0 +1,57 @@
+module F = Vhd_format.F.From_file(Vhd_format_lwt.IO)
+
+open Lwt.Infix
+
+type extent = {
+  flags : int32;
+  length : int64;
+} [@@deriving rpc]
+
+(* The flags returned for the base:allocation NBD metadata context are defined here:
+   https://github.com/NetworkBlockDevice/nbd/blob/extension-blockstatus/doc/proto.md#baseallocation-metadata-context *)
+let flag_hole = 1l
+let flag_zero = 2l
+type extent_list = extent list [@@deriving rpc]
+
+let get_extents_json ~server ~export_name =
+  let extent_reader = "/opt/xensource/libexec/get_nbd_extents.py" in
+  Lwt_process.pread
+    ("", [|extent_reader; "--path"; server; "--exportname"; export_name|])
```

Comment:
I've realized that this does not raise an exn but simply returns an empty string when the script does not exist, that's not good at all